sdk/identity/identity/src/credentials/defaultAzureCredential.ts

```diff
@@ -56,17 +47,35 @@ export interface DefaultAzureCredentialOptionsWithResourceId extends TokenCreden
 /**
  * Provides options to configure the {@link DefaultAzureCredential} class.
  */
-export type DefaultAzureCredentialOptions =
-  | DefaultAzureCredentialClientIdOptions
-  | DefaultAzureCredentialOptionsWithResourceId;
+export interface DefaultAzureCredentialOptions extends TokenCredentialOptions {
+  /**
+   * Optionally pass in a Tenant ID to be used as part of the credential.
+   * By default it may use a generic tenant ID depending on the underlying credential.
+   */
+  tenantId?: string;
+}
 
 /**
  * The type of a class that implements TokenCredential and accepts
- * `DefaultAzureCredentialOptions`.
+ * {@link DefaultAzureCredentialOptions}.
  */
 interface DefaultCredentialConstructor {
   new (options?: DefaultAzureCredentialOptions): TokenCredential;
 }
+/**
+ * The type of a class that implements TokenCredential and accepts
+ * {@link DefaultAzureCredentialResourceIdOptions}.
+ */
+interface DefaultCredentialConstructor {
+  new (options?: DefaultAzureCredentialResourceIdOptions): TokenCredential;
+}
+/**
+ * The type of a class that implements TokenCredential and accepts
+ * {@link DefaultAzureCredentialClientIdOptions}.
+ */
+interface DefaultCredentialConstructor {
+  new (options?: DefaultAzureCredentialClientIdOptions): TokenCredential;
+}
```

Comment:
Does this have to be two separate interfaces? There's no real problem with it since they're merged, but can you decalre them side by side and document the constructors differently rather than the whole interface?

Reply:
if merging them still allows the editor to not suggest the other case when one of the paths is taken, then I agree

Reply:
No apparent difference. This makes sense given that TypeScript merges interface declarations implicitly anyways.

https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgILIN4Chm+XALmQGcwpQBzAbiwF8stRJZEUAhTHPAIyJAFcAtt2g16WMAE8ADigCSIUnBBJkAXky0ajcNHiqAYgHsjnPMhAQA7sgAUhNAEoiCpSohiGTPa2THT2OaWVra8yGzOyK5gykieOsz67HBQZnjB9kSokdGxHly4GWERLoox7vEAJhAIADYpKAhGZX4mRP401XUNyE0tbClEA1DaAPSjyABypnDS0g3gyJXAMDDQEO5YfaTIMCbqFtatRrYYyGEAjAAMyLSONNtg5ykHweEpp-hEAESQpN93KhAA

Reply:
alright! then let’s wait for Karishma to follow up 🙂 thank you, Will!